recipe/recipe.yaml

```diff
@@ -0,0 +1,152 @@
+context:
+  version: "6_5"
+  pkg_version: ${{ version | split("_") | join(".") }}
+
+package:
+  name: ncurses
+  version: ${{ pkg_version }}
+
+source:
+  url: https://github.com/ThomasDickey/ncurses-snapshots/archive/refs/tags/v${{ version }}.tar.gz
+  sha256: 1600526b279b0acf42432fe31f6dfa3824a4da4262168bf8b3637660ff8c07c7
+
+build:
+  skip:
+    - win
+  number: 2
+
+requirements:
+  build:
+    - ${{ stdlib('c') }}
+    - ${{ compiler('c') }}
+    - ${{ compiler('cxx') }}
+    - pkg-config
+    - make
+    - libtool
+    - coreutils
+    - ncurses  # [build_platform != target_platform]
+
+  host:
+    # Need a separate host section so that ncurses doesn't remove -L$PREFIX/lib
+    - libcxx
+
+  # A C++ compiler is needed so that the cursesapp.h file is installed,
+  # but the library does not need libstdc++/libc++
+  ignore_run_exports:
+    from_package: 
+      - ${{ compiler('cxx') | split | first }}
```

Comment:
How does rattler handle ignored run-exports that come twice from different packages? In conda, with both a c & cxx compiler,
```yaml
ignore_run_exports_from:
  - {{ compiler("cxx") }}
```
will not remove the `libgcc` run-export, because it also comes from `{{ compiler("c") }}`. I think we should do the same here.

Relying on whatever happens to be the first run-export of the cxx compiler sounds way too fragile.

Reply:
This just splits the matchspec so that it reads `clang` instead of `clang 17` or so.

Reply:
I'm not sure I understand. For the `from_package` functionality, no further processing should be necessary.

IOW, if I have package `X` in my requirements but want to ignore its run-export, all that should be required is to put `X` under `ignore_run_exports.from_package`; that should be it, even if `X = {{ compiler("cxx") }}`.

Reply:
Not sure how conda-build does this; I can understand that populating the compiler jinja literally would give you an extra version that's kinda non-sensical in the `ignore_run_exports.from_package`; but it might be easier to allow that and then just ignore version numbers when processing the REs (or require that they match?)

Reply:
Yeah, it's something we can fix in rattler-build. Currently we expect a `PackageName` type here - which can only be a package name. However, `${{ compiler('cxx') }}` evaluates to a MatchSpec (`clang 17`) and that does not parse properly. For the moment, we can fix it as indicated.

Reply:
https://github.com/prefix-dev/rattler-build/issues/1264

Reply:
```suggestion
      # need to remove version pin, c.f. https://github.com/prefix-dev/rattler-build/issues/1264
      - ${{ compiler('cxx') | split | first }}
```